pkg/validator/validators.go

```diff
@@ -360,22 +391,53 @@ func validateMRCProvider(mrc *configv1alpha2.MeshRootCertificate) error {
 	return nil
 }
 
-// checkForExistingActiveMRC returns true if there are active MRCs in addition to the specified MRC
+// mrcWithMatchingIntentExists returns true if there are active MRCs in addition to the specified MRC
 // Returns false if there are no active MRCs or if the specified mrc will be the only active MRC
-func (kc *validator) checkForExistingActiveMRC(mrc *configv1alpha2.MeshRootCertificate) (bool, error) {
+func mrcWithMatchingIntentExists(mrcs []*configv1alpha2.MeshRootCertificate, mrc *configv1alpha2.MeshRootCertificate) bool {
+	for _, m := range mrcs {
+		if m.Spec.Intent == mrc.Spec.Intent && m.Name != mrc.Name {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (kc *validator) getUseableMRCs() ([]*configv1alpha2.MeshRootCertificate, error) {
```

Comment:
Should we use this method in place of `filterOutInactiveMRCs`? This code might be more readable.

Reply:
neither of this should be needed after #5226. we can address it there.